contracts/DAORegistryScheme.sol

```diff
@@ -0,0 +1,181 @@
+            bytes memory genericCallReturnValue;
+            bool success;
+            ControllerInterface controller = ControllerInterface(avatar.owner());
+            (success, genericCallReturnValue) =
+                controller.genericCall(address(params.daoRegistry), proposal.callData, avatar, proposal.value);
+            require(success, "proposal external call cannot be executed");
+            emit ProposalExecuted(address(avatar), _proposalId, _param, genericCallReturnValue);
+        }
+
+        return true;
+    }
+
+    /**
+    * @dev hash the parameters, save them if necessary, and return the hash value
+    */
+    function setParameters(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        DAORegistry _daoRegistry
+    ) public returns(bytes32)
+    {
+        bytes32 paramsHash = getParametersHash(_voteParams, _intVote, _daoRegistry);
+        parameters[paramsHash].voteParams = _voteParams;
+        parameters[paramsHash].intVote = _intVote;
+        parameters[paramsHash].daoRegistry = _daoRegistry;
+        return paramsHash;
+    }
+
+    function getParametersHash(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        DAORegistry _daoRegistry
+    ) public pure returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(_voteParams, _intVote, address(_daoRegistry)));
+    }
+
+    /**
+    * @dev create a proposal to register a DAO in the registry
+    * @param _avatar the address of the organization owning the registry
+    * @param _proposedAvatarName the name of the organization we want to add to the registry
+    * @param _proposedAvatar the organization we want to add to the registry
+    * @param _value value(ETH) to transfer with the call
+    * @return a proposal Id
+    */
+    function proposeToAddDAO(
+        Avatar _avatar,
+        string memory _proposedAvatarName,
+        Avatar _proposedAvatar,
+        uint256 _value
+    ) public returns(bytes32)
+    {
+        Parameters memory params = parameters[getParametersFromController(_avatar)];
+
+        bytes32 proposalId = params.intVote.propose(
+            2,
+            params.voteParams,
+            msg.sender,
+            address(_avatar)
+        );
+
+        bytes memory callData =
+            abi.encodeWithSignature("register(address, string memory)", address(_proposedAvatar), _proposedAvatarName);
+
+        DAORegistryProposal memory proposal = DAORegistryProposal({
+            callData: callData,
+            value: _value
+        });
+
+        emit NewDAOProposal(
+            address(_avatar),
+            proposalId,
+            address(_proposedAvatar),
+            _proposedAvatarName
+        );
+
+        organizationsProposals[address(_avatar)][proposalId] = proposal;
+        proposalsInfo[address(params.intVote)][proposalId] = ProposalInfo({
+            blockNumber:block.number,
+            avatar:_avatar
+        });
+
+        return proposalId;
```

Comment:
how about defining proposalId as the function return value (`public returns(bytes32 proposalId) `) and remove this return ?